bluepysnap/nodes.py

```diff
@@ -32,12 +32,201 @@
+            >>> nodes.ids(group={Node.LAYER: 2})  # returns CircuitNodeIds matching layer==2
+            >>> nodes.ids(group={Node.LAYER: [2, 3]})  # returns CircuitNodeIds with layer in [2,3]
+            >>> nodes.ids(group={Node.X: (0, 1)})  # returns CircuitNodeIds with 0 < x < 1
+            >>> # returns CircuitNodeIds matching one of the queries inside the 'or' list
+            >>> nodes.ids(group={'$or': [{ Node.LAYER: [2, 3]},
+            >>>                          { Node.X: (0, 1), Node.MTYPE: 'L1_SLAC' }]})
+            >>> # returns CircuitNodeIds matching all the queries inside the 'and' list
+            >>> nodes.ids(group={'$and': [{ Node.LAYER: [2, 3]},
+            >>>                           { Node.X: (0, 1), Node.MTYPE: 'L1_SLAC' }]})
+        """
+        str_type = "<U{}".format(max(len(pop) for pop in self.population_names))
+        if isinstance(group, CircuitNodeIds):
+            diff = np.setdiff1d(group.get_populations(unique=True), self.population_names)
+            if diff.size != 0:
+                raise BluepySnapError("Population {} does not exist in the circuit.".format(diff))
+        ids = np.array([], dtype=np.int64)
+        populations = np.array([], dtype=str_type)
+        for name, pop in self.items():
+            pop_ids = pop.ids(group=group, raise_missing_property=False)
+            pops = np.array(np.full_like(pop_ids, fill_value=name, dtype=str_type))
+            ids = pop_ids if ids.size == 0 else np.concatenate([ids, pop_ids])
+            populations = pops if populations.size == 0 else np.concatenate([populations, pops])
+        return CircuitNodeIds.create_ids(populations, ids)
+
+    def get(self, group=None, properties=None):
+        """Node properties as a pandas DataFrame.
+
+        Args:
+            group (CircuitNodeIds/int/sequence/str/mapping/None): Which nodes will have their
+                properties returned depends on the type of the ``group`` argument:
+                See :py:class:`~bluepysnap.nodes.Nodes.ids`.
+
+            properties (list): If specified, return only the properties in the list.
+                Otherwise return all properties.
+
+        Returns:
+            pandas.DataFrame: Return a pandas DataFrame indexed by NodeCircuitIds containing the
+                properties from ``properties``.
+
+        Notes:
+            The NodePopulation.property_names function will give you all the usable properties
+            for the `properties` argument.
+        """
+        ids = self.ids(group)
+        if properties is None:
+            properties = self.property_names
+        properties = utils.ensure_list(properties)
+        res = pd.DataFrame(index=ids.index, columns=properties)
+        for name, pop in self.items():
+            global_pop_ids = ids.filter_population(name)
+            pop_ids = global_pop_ids.get_ids()
+            pop_properties = set(properties) & pop.property_names
+            for prop in pop_properties:
```

Comment:
Why `for`? Can you put a comment why we can't set them at once?

Reply:
Because not all properties are from the same type. So you cannot use a numpy array to set them directly. 
If you use a dataframe then it checks the indices and try to match them. But here they are different for the NodePopulation and the Nodes get functions.